### PR TITLE
fixes #203 enable configuration from dbt_project.yml

### LINKED ISSUE #203: Update dbt select/exclude documentation
This [section](https://github.com/astronomer/astronomer-cosmos/blob/main/docs/dbt/configuration.rst?plain=1#L32-L35) suggests that you should structure your select or exclude like this.

```python3
select={"configs": "tags:daily"}
```
This too did not work but at least did not result in no tasks being returned.
```python3
select={"tags": "daily"}
```
I'm setting my tags in dbt_project.yml, not schema.yml or directly inside the models like below.
Is this supported?
```yaml
models:
  jaffle_shop:
      staging:
        +tags:
          - hourly
      modelling:
        +tags:
          - daily
      mart:
        +tags:
          - daily
```

### PR DESCRIPTION
**Pull request by MikeWallis42**
fixes #203 add support for selectors defined in dbt_project.yml

### CODE REVIEW

File: tests/dbt/parser/test_project.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+            "tags: confidential",
+        }
+    )
+    merged_config = sql_config + schema_config + project_config
+    assert merged_config.config_selectors == {
+        "materialized:incremental",
+        "schema:jaffle_shop",
+        "tags:finance",
+        "tags:hourly",
+        "tags: confidential",
```
Comment (jlaneve) [inline tests/dbt/parser/test_project.py:27]: does this need a space?

File: tests/dbt/parser/test_project.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        config_selectors={"materialized:incremental", "tags:finance"}
+    )
+    schema_config = DbtModelConfig(
+        config_selectors={"materialized:table", "schema:jaffle_shop", "tags:hourly"}
+    )
+    project_config = DbtModelConfig(
+        config_selectors={
+            "materialized:view",
+            "schema:my_project",
+            "tags: confidential",
```
Comment (jlaneve) [inline tests/dbt/parser/test_project.py:18]: does this need a space?

File: tests/dbt/parser/test_project.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    dbt_model = DbtModel("jaffle_shop_revenue", sql_model)
+    dbt_model_config = dbt_model.config
+    assert dbt_model_config.config_selectors == {
+        "materialized:incremental",
+        "schema:jaffle_shop",
+        "tags:daily",
+    }
+
+
+# TODO sources test
```
Comment (jlaneve) [inline tests/dbt/parser/test_project.py:81]: is this a TODO for this PR or a follow-on?

File: tests/dbt/parser/test_project.py (hunk 1 above)
Comment (MikeWallis42) [reply tests/dbt/parser/test_project.py:27]: Will correct 👍

File: tests/dbt/parser/test_project.py (hunk 2 above)
Comment (MikeWallis42) [reply tests/dbt/parser/test_project.py:18]: Copy/paste error

**Comment by MikeWallis42** [comment]
Hi @jlaneve, I've opened up a new PR rather than trying to work through a rebase #253
I'll close this one down now.

File: tests/dbt/parser/test_project.py (hunk 3 above)
Comment (jackwelty) [reply tests/dbt/parser/test_project.py:81]: Is the testing of sources supported by cosmos?

**Pull request closed without merging**